pkg/kv/kvserver/raft_transport.go

```diff
@@ -235,6 +383,37 @@ func (t *RaftTransport) getHandler(storeID roachpb.StoreID) (RaftMessageHandler,
 func (t *RaftTransport) handleRaftRequest(
 	ctx context.Context, req *kvserverpb.RaftMessageRequest, respStream RaftMessageResponseStream,
 ) *kvpb.Error {
+	for i := range req.AdmittedRaftLogEntries {
+		// Process any flow tokens that were returned over the RaftTransport. Do
+		// this first thing, before these requests enter the receive queues
+		// which could drop them if full and cause token leaks, or bail after
+		// processing the raft heartbeats. See I8 from kvflowcontrol/doc.go.
+		admittedEntries := req.AdmittedRaftLogEntries[i]
+		handle, found := t.kvflowControl.handles.Lookup(admittedEntries.RangeID)
```

Comment:
Same response as elsewhere. BTW, the admittedEntries.StoreID is _not_ what we want to pass down here. That field has the following comment:
```
// StoreID on which this raft log entry was admitted.
```

So that's the store ID on the follower node, below-raft. What we want to pass to Lookup is the store ID on the proposer node, above-raft where the tokens were deducted.